android/engine/src/main/java/org/smartregister/fhircore/engine/util/FileUtil.kt

```diff
@@ -87,4 +89,45 @@ class FileUtil {
     val ASSET_BASE_PATH_RESOURCES =
       (System.getProperty("user.dir") + File.separator + "src" + File.separator)
   }
+
+  fun writeFileOnInternalStorage(
+    context: Context,
+    fileName: String?,
+    body: String?,
+    dirName: String
+  ) {
+
+    val dir = File(context.getFilesDir(), dirName)
+    if (!dir.exists()) {
+      dir.mkdir()
+    }
+    try {
+      val file = File(dir, fileName)
+      val writer = FileWriter(file)
+      writer.append(body)
+      writer.flush()
+      writer.close()
+    } catch (e: Exception) {
+      e.printStackTrace()
+    }
+  }
+
+  fun readFileFromInternalStorage(context: Context, fileName: String?, dirName: String): String {
+    val dir = File(context.getFilesDir(), dirName)
+    val sb = java.lang.StringBuilder()
+    try {
+      val file = File(dir, fileName)
+      val fis = FileInputStream(file)
+      val isr = InputStreamReader(fis)
+      val bufferedReader = BufferedReader(isr)
+      var line: String?
+      while (bufferedReader.readLine().also { line = it } != null) {
+        sb.append(line)
+      }
+      fis.close()
+    } catch (e: Exception) {
```

Comment:
can we catch a tighter exception? Don't we use Timber for reporting instead of print?

Reply:
Use the Kotlin [use](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html) function for the classes that implement [Closeable](https://docs.oracle.com/javase/8/docs/api/java/io/Closeable.html)

Reply:
@pld This addressed by implementing @ellykits suggestion